### DISCUSSION TITLE
stream_with_context and unhandled exceptions

### DISCUSSION THREAD

**Discussion by wrkhenddher** [pallets/flask#5318] [Q&A]
Hello folks,

Within the context of [`@stream_with_context`](https://flask.palletsprojects.com/en/3.0.x/patterns/streaming/#streaming-contents) and this [old flask issue](https://github.com/pallets/flask/issues/2992), I have a question:

Is there any hook or facility to finish streaming content once an exception is raised within the generator?

Our current solution is to wrap the generator with a `try/except` block and yield "Output truncated - Exception..." when an exception is caught.

```python

# Refer to https://github.com/pallets/flask/issues/2992

@app.route('/streamingerror')
def streaming_error_route():
    def inner_gen():
        try: 
            yield 'hello'
            raise Exception('error from streaming_error_route')
            yield 'bye'
        except Exception as e:
            logging.exception("Exception in inner_gen")
            yield f"Output truncated - {e}"
            raise e
    return Response(stream_with_context(inner_gen()))
```

The main problems are:
1. The `HTTP/1.0 200 OK` header has been sent but content is truncated due to the exception.
2. The `Content-Length` header is absent.
3. Without the `"Output truncated - Exception..."`, there is no way to know there was such error.
4. The custom error handler is never called (as pointed by the old flask issue).

Perhaps the more fundamental question is:

How should one deal with abrupt terminations within a generator wired to `@stream_with_context`?

Bit of extra context:

```shell
❯ python --version
Python 3.8.16

❯ pip freeze --all
blinker==1.7.0
click==8.1.7
Flask==3.0.0
importlib-metadata==6.8.0
itsdangerous==2.1.2
Jinja2==3.1.2
MarkupSafe==2.1.3
pip==22.0.4
setuptools==56.0.0
Werkzeug==3.0.1
zipp==3.17.0
```

**Comment by williamswhy** [comment]
Hi, did you find any workaround? I am having the same issue. Thanks!
